sphinx_gallery/tests/test_gen_rst.py

```diff
@@ -104,3 +110,37 @@ def test_md5sums():
         assert_false(sg.check_md5sum_change(f.name))
 
     os.remove(f.name + '.md5')
+
+
+def test_pattern_matching():
+    """Test if only examples matching pattern are executed"""
```

Comment:
aren't all the other tests using docstrings like this? I suppose that would be a different PR if you wanted to fix that?